Src/BootCamp.Chapter/Credentials.cs

```diff
@@ -1,22 +1,64 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Linq;
+using System.Text;
+
+namespace BootCamp.Chapter
 {
     // TODO: make a struct and add validation and other needed methods (if needed)
     public class Credentials
     {
-        public string Username;
-        public string Password;
+        public readonly string Username;
+        public readonly string Password;
 
         public Credentials(string username, string password)
         {
+            if (string.IsNullOrEmpty(username))
+            {
+                throw new ArgumentException("You must provide username");
+            }
+            if (string.IsNullOrEmpty(password))
+            {
+                throw new ArgumentException("You must provide password");
+            }
             Username = username;
-            Password = password;
+            var encodedPassword = new EncodedPassword(password);
+            Password = encodedPassword.PasswordAfterEncoding;
         }
 
         // TODO: Implement properly.
         public static bool TryParse(string input, out Credentials credentials)
         {
             credentials = default;
-            return false;
+            if (string.IsNullOrEmpty(input)) return false;
+
+            var CredentialsParts = input.Split(",");
+            if (CredentialsParts.Length != 2) return false;
+
+            if (string.IsNullOrEmpty(CredentialsParts[0]) || string.IsNullOrEmpty(CredentialsParts[1])) return false;
+
+            credentials = new Credentials(CredentialsParts[0], CredentialsParts[1]);
+            return true;
+        }
+        public readonly struct EncodedPassword
```

Comment:
I think this nested class is not needed. Essentially, not having it is the same as just having a string. Right now, it's a roundtrip with no value.

Reply:
All that is needed is already encapsulated in Credentials class.